src/workerd/jsg/jsg.h

```diff
@@ -2042,6 +2042,36 @@ class JsMessage;
 
 class DOMException;
 
+// RAII class to adjust the amount of external memory attributed to an isolate.
+// The adjustment will be automatically decremented when the object is destroyed.
+// The allocation amount can be adjusted up or down during the lifetime of an object.
+class ExternalMemoryAdjuster final {
```

Comment:
Bikeshedding: Maybe `ExternalMemoryAdjustment` is a more natural name? The object itself represents an adjustment.